doc/source/reference/arrays.scalars.rst

```diff
@@ -169,6 +174,25 @@ Inexact types
    :members: __init__
    :exclude-members: __init__
 
+.. note::
+
+   The printing behavior of inexact scalars is to use as few digits as possible
+   to display the result unambiguously. This means that equal values at
+   different precisions will display differently::
+
+       >>> f16 = np.float16("0.1")
+       >>> f32 = np.float32(f16)
+       >>> f64 = np.float64(f32)
+       >>> f16 == f32 == f64
+       True
+       >>> f16, f32, f64
+       (0.1, 0.099975586, 0.0999755859375)
+
+   Note that none of these floats hold the exact value :math:`\frac{1}{10}`;
+   ``f16`` prints as ``0.1`` because it is as close to that value as possible,
+   whereas the other types do not as they have more precision and therefore have
+   closer values.
```

Comment:
Thoughts on extending this to:

```suggestion
   Note that none of these floats hold the exact value :math:`\frac{1}{10}`;
   ``f16`` prints as ``0.1`` because it is as close to that value as possible,
   whereas the other types do not as they have more precision and therefore have
   closer values.
   
   Conversely, the same value approximated at different precisions may compare
   unequal despite printing identically:
   
       >>> f16 = np.float16("0.1")
       >>> f32 = np.float32("0.1")
       >>> f64 = np.float64("0.1")
       >>> f16 == f32 == f64
       False
       >>> f16, f32, f64
       (0.1, 0.1, 0.1)
```

Reply:
I think this is good to add, but I might tweak the language:

> Note that none of these floats hold the exact fraction :math:`\frac{1}{10}`, which cannot be represented in binary with finite precision.  The ``f16`` scalar prints as ``0.1`` because it is the ``f16`` value which is closest to the decimal value 0.1 among all ``f16`` values,  whereas the ``f32`` and ``f64`` scalars do not as their datatypes have more precision and therefore have other values closer to decimal 0.1.

> Conversely,  floating-point scalars of different precisions which approximate the same decimal value may compare unequal despite printing identically:

Reply:
Perhaps consider adding:
```
>>> np.format_float_positional(np.float16('0.1'), unique=False, precision=40)
'0.0999755859375000000000000000000000000000'
>>> np.format_float_positional(np.float32('0.1'), unique=False, precision=40)
'0.1000000014901161193847656250000000000000'
>>> np.format_float_positional(np.float64('0.1'), unique=False, precision=40)
'0.1000000000000000055511151231257827021182'
```

Reply:
I think that last comment probably belongs in the docs for `format_float_positional`.

Reply:
I've not taken your changes because they were awkard to edit in - would you mind re-commenting as suggested changes?